config/routes.rb

```diff
@@ -3,4 +3,20 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+    # All users page
+    get '/users', to: 'users#index', as: 'users'
+
+    # Single user page
+    get '/users/:id', to: 'users#show', as: 'user'
+  
+    # All posts by a given user page
+    get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  
+    # Single post page
+    get '/posts/:id', to: 'posts#show', as: 'post'
```

Comment:
- [ ] Good job working on routes 👍, however,  the use of `:id` for both users and posts in your routes can lead to unexpected behavior. This could cause conflicts, as Rails wouldn't know whether to route the request to the UsersController or the PostsController. To make your routes more clear and avoid potential conflicts, I recommend using different parameter names for the user's ID.

````ruby
get '/users/:user_id/posts', to: 'posts#index'
get '/users/:user_id/posts/:id', to: 'posts#show'
````

- [ ] [OPTIONAL] I think it would be better to use [resource routing](https://guides.rubyonrails.org/v3.2.2/routing.html#resource-routing-the-rails-default) for defining routes. It not only simplifies your code but also aligns with Rails conventions, making your application more maintainable and developer-friendly.
````ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
````

Reply:
Changed, Thank you for your examples.